lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -80,13 +73,31 @@ def declare_suspicious_replicas_bad(once: bool = False, younger_than: int = 3, n
     :returns: None
     """
 
+    run_daemon(
+        once=once,
+        graceful_stop=GRACEFUL_STOP,
+        executable=argv[0],
+        logger_prefix='replica_recoverer',
+        partition_wait_time=1,
+        sleep_time=sleep_time,
+        run_once_fnc=functools.partial(
+            run_once,
+            younger_than=younger_than,
+            nattempts=nattempts,
+            vos=vos,
+            limit_suspicious_files_on_rse=limit_suspicious_files_on_rse,
+            active_mode=active_mode
+        ),
+    )
+
+
+def run_once(heartbeat_handler: Any, younger_than: int, nattempts: int, vos: Optional[List[str]], limit_suspicious_files_on_rse: int, active_mode: int, **_kwargs):
```

Comment:
The return type is either `None` or `bool`. Could you converge to one and add a hint?

Reply:
Sorry, maybe I wasn’t clear enough. You have two return statements in `run_once()`, each with a different type:
https://github.com/rucio/rucio/blob/d189dd1df70546576b690135db08cf8ba7959d80/lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py#L127
https://github.com/rucio/rucio/blob/d189dd1df70546576b690135db08cf8ba7959d80/lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py#L411